logging/telemetryspec/metric.go

```diff
@@ -57,7 +59,17 @@ type AssembleBlockStats struct {
 	ProcessingTime            transactionProcessingTimeDistibution
 	BlockGenerationDuration   uint64
 	TransactionsLoopStartTime int64
-	HasStateProofNextRound    uint64 // next round for which state proof if expected. Set only if this block has a state proof
+	StateProofNextRound       uint64 // next round for which state proof if expected
+	StateProofStats           StateProofStats
+}
+
+// StateProofStats is the set of stats captured when a StateProof is present in the assembled block
+type StateProofStats struct {
+	StateProofProvenWeight    uint64
+	StateProofSignedWeight    uint64
+	StateProofNumReveals      int
+	NumberOfPositionsToReveal int
```

Comment:
If we need the prefix `stateproof` for each field I guess `NumberOfPositionsToReveal` should follow this pattern as well.